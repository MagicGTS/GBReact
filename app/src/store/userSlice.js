import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  isAuth: false,
  user: null,
  error: null,
  loading: false,
};
export const userSlice = createSlice({
  name: "user",
  initialState: initialState,
  reducers: {
    getAuthRequest: (state) => {
      return {
        ...state,
        loading: true,
        isAuth: false,
      };
    },
    getAuthSuccess: (state, action) => {
      return {
        ...state,
        user: action.payload,
        loading: false,
        isAuth: true,
      };
    },
    getAuthFailure: (state, action) => {
      return {
        ...state,
        error: action.payload,
        loading: false,
        isAuth: false,
      };
    },
  },
});
export function selectUser() {
  return (state) => state.user.name
}
export function selectIsAuth() {
  return (state) => state.user.isAuth
}
export function selectUserError() {
  return (state) => state.user.error
}
export function selectUserLoading() {

  return (state) => state.user.loading
}
export const { getAuthRequest, getAuthSuccess, getAuthFailure } = userSlice.actions
export default userSlice.reducer;
