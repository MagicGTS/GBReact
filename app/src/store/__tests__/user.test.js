import { getAuthSuccess } from '../userSlice.js'
import userReducer from '../userSlice.js'
describe('snapshot tests', () => {
    it('returns correct state after SET_NAME action', () => {
        const userStore = userReducer(undefined, getAuthSuccess('test'));
        expect(userStore).toMatchSnapshot();
        });
    // more tests...
});
