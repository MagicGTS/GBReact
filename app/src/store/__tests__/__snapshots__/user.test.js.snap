// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests returns correct state after SET_NAME action 1`] = `
Object {
  "error": null,
  "isAuth": true,
  "loading": false,
  "user": "test",
}
`;
